src/infill/ZigzagConnectorProcessor.cpp

```diff
@@ -134,21 +134,38 @@ void ZigzagConnectorProcessor::registerPolyFinished()
 }
 
 
-void ZigzagConnectorProcessor::addZagConnector(const std::vector<Point>& points, bool is_endpiece)
+void ZigzagConnectorProcessor::addZagConnector(std::vector<Point>& points, bool is_endpiece)
 {
     // don't include the last line yet
     if (points.size() >= 3)
     {
         for (size_t point_idx = 1; point_idx <= points.size() - 2; ++point_idx)
         {
-            addLine(points[point_idx - 1], points[point_idx]);
+            Point* from = &points[point_idx - 1];
+            Point* to = &points[point_idx];
+            if (!mergePointsByThreshold(from, to))
+                addLine(*from, *to);
         }
     }
     // only add the last line if:
     //  - it is not an end piece, or
     //  - it is an end piece and "connected end pieces" is enabled
     if ((!is_endpiece || (is_endpiece && this->connected_endpieces)) && points.size() >= 2)
     {
-        addLine(points[points.size() - 2], points[points.size() - 1]);
+        Point* from = &points[points.size() - 2];
+        Point* to = &points[points.size() - 1];
+        if (!mergePointsByThreshold(from, to))
+            addLine(*from, *to);
     }
 }
+
+
+bool ZigzagConnectorProcessor::mergePointsByThreshold(Point* first_point, Point* second_point, int threshold)
+{
+    if (vSize2(*first_point - *second_point) < threshold*threshold)
```

Comment:
Some description, why do we check? threshold*threshold

Reply:
Because vSize2() returns the square of length of the vector. We could use vSize() which returns the length of the vector but it's quicker to do a multiply than a square root.

Reply:
I think it's fair to say that using squared distances in the engine code is well known technique and so I would not expect this to require a comment.

Reply:
Yes, as Mark pointed, this is for avoid calculating the square root, and because it was the same strategy followed in PR #614.
I can add a comment anyway.

Reply:
Now it is clear, Personally, I try to follow the strategy to  comment "Why, I do check".